realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -938,551 +655,57 @@ public void findFirstAsync_forceLoad() throws Throwable {
-
-                        assertEquals("data 5", realmResults2.get(0).getColumnString());
-                        assertEquals(0, realmResults2.get(0).getColumnLong());
-
-                        assertEquals("data 0", realmResults2.get(11).getColumnString());
-                        assertEquals(3, realmResults2.get(11).getColumnLong());
-
-                        assertEquals("data 1", realmResults2.get(5).getColumnString());
-                        assertEquals(Math.PI, realmResults2.get(5).getColumnDouble(), 0.000000000001D);
-
-                        signalCallbackDone.run();
-                        break;
-                }
-            }
-        });
-    }
-
-    // make sure the notification listener does not leak the enclosing class
-    // if unregistered properly.
-    @Test
-    @RunTestInLooperThread
-    public void listenerShouldNotLeak() {
-        populateTestRealm(looperThread.realm, 10);
-
-        // simulate the ActivityManager by creating 1 instance responsible
-        // of attaching an onChange listener, then simulate a configuration
-        // change (ex: screen rotation), this change will create a new instance.
-        // we make sure that the GC enqueue the reference of the destroyed instance
-        // which indicate no memory leak
-        MockActivityManager mockActivityManager =
-                MockActivityManager.newInstance(looperThread.realm.getConfiguration());
-
-        mockActivityManager.sendConfigurationChange();
-
-        assertEquals(1, mockActivityManager.numberOfInstances());
-        // remove GC'd reference & assert that one instance should remain
-        Iterator<Map.Entry<WeakReference<RealmResults<?>>, RealmQuery<?>>> iterator =
-                looperThread.realm.handlerController.asyncRealmResults.entrySet().iterator();
-        while (iterator.hasNext()) {
-            Map.Entry<WeakReference<RealmResults<?>>, RealmQuery<?>> entry = iterator.next();
-            RealmResults<?> weakReference = entry.getKey().get();
-            if (weakReference == null) {
-                iterator.remove();
-            }
-        }
-
-        assertEquals(1, looperThread.realm.handlerController.asyncRealmResults.size());
-        mockActivityManager.onStop();// to close the Realm
-        looperThread.testComplete();
-    }
+        });
+    }
 
     @Test
     @RunTestInLooperThread
     public void combiningAsyncAndSync() {
         populateTestRealm(looperThread.realm, 10);
 
-        Realm.asyncTaskExecutor.pause();
         final RealmResults<AllTypes> allTypesAsync = looperThread.realm.where(AllTypes.class).greaterThan("columnLong", 5).findAllAsync();
         final RealmResults<AllTypes> allTypesSync = allTypesAsync.where().greaterThan("columnLong", 3).findAll();
 
         assertEquals(0, allTypesAsync.size());
-        assertEquals(6, allTypesSync.size());
+        assertEquals(4, allTypesSync.size()); // columnLong > 5 && columnLong > 3
```

Comment:
using `where()` on an unloaded async results forgot the query conditions?

sounds like a bug that was never found